cli/src/commands/node/mod.rs

```diff
@@ -216,7 +219,27 @@ impl Node {
                 node::config::DEVNET_CONFIG.clone(),
             ),
         };
-        let mut node_builder: NodeBuilder = NodeBuilder::new(None, daemon_conf, genesis_conf);
+
+        let custom_rng_seed = match self.rng_seed {
+            None => None,
+            Some(v) => match hex::decode(v)
+                .map_err(anyhow::Error::from)
+                .and_then(|bytes| {
+                    <[u8; 32]>::try_from(bytes.as_slice()).map_err(anyhow::Error::from)
+                }) {
+                Ok(v) => Some(v),
+                Err(err) => {
+                    node::core::error!(
+                        node::core::log::system_time();
+                        summary = "bad rng seed",
+                        err = err.to_string(),
+                    );
+                    return Err(err);
+                }
+            },
+        };
+        let mut node_builder: NodeBuilder =
+            NodeBuilder::new(custom_rng_seed, daemon_conf, genesis_conf);
```

Comment:
Only things that are required should be passed via `NodeBuilder::new`. Since custom rng is optional, please remove it and use similar pattern instead:
```
if let Some(sec_key) = self.p2p_secret_key {
            node_builder.p2p_sec_key(sec_key);
}
```